### PR TITLE
PathProvider

### PR DESCRIPTION
**Pull request by Slickoo**
IPathProvider interface.
LocalPathProvider class.
Unit tests.

### CODE REVIEW

File: NPlaylist/Tests/NPlaylist.Business.Tests/Providers/PathProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        [Fact]
+        public void BuildPath_PathContainsFileExtension_True()
+        {
+            var pathMock = Substitute.For<IPathWrapper>();
+            pathMock.GetExtension("test.mp3").Returns(".mp3");
+            var guidMock = Substitute.For<IGuidWrapper>();
+            guidMock.NewGuid().Returns(Guid.Parse("c74aea2d-4830-4987-932d-b21ddab417cd"));
+            var sut = new LocalPathProvider("webRootPath", pathMock, guidMock);
+
+            var expectedResult = "webRootPath/Files/mp3/c74aea2d-4830-4987-932d-b21ddab417cdtest.mp3";
```
Comment (TermanEmil) [inline NPlaylist/Tests/NPlaylist.Business.Tests/Providers/PathProviderTests.cs]: Isn't this too rigid?

File: NPlaylist/NPlaylist.Business/Providers/LocalPathProvider.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+            var fileExtension = _pathWrapper.GetExtension(fileName);
+
+            if (string.IsNullOrWhiteSpace(fileExtension))
+            {
+                throw new ArgumentException(nameof(fileExtension));
+            }
+
+            fileExtension = fileExtension.Replace(".", "");
+            var path = $"{_appEnvironmentPath}/Files/{fileExtension}/{_guidWrapper.NewGuid().ToString()}{fileName}";
```
Comment (aivascu) [inline NPlaylist/NPlaylist.Business/Providers/LocalPathProvider.cs]: Do path concatenation using `Path.Combine(string[])`
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=netframework-4.7.2

File: NPlaylist/Tests/NPlaylist.Business.Tests/Providers/PathProviderTests.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+            sut.BuildPath("test.mp3").Should().Be(expectedResult);
+        }
+
+        [Fact]
+        public void BuildPath_PathShouldContainExtension_True()
+        {
+            var pathMock = Substitute.For<IPathWrapper>();
```
Comment (aivascu) [inline NPlaylist/Tests/NPlaylist.Business.Tests/Providers/PathProviderTests.cs]: Reduce the code duplication using the builder pattern

File: NPlaylist/Tests/NPlaylist.Business.Tests/Providers/PathProviderTests.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+            sut.BuildPath("test.mp3").Should().NotBeNullOrEmpty();
+        }
+        [Fact]
+        public void BuildPath_PathProviderShouldReturnString_True()
+        {
+            var sut = new LocalPathProviderBuilder()
+                .Build();
+
+            sut.BuildPath("test.mp3").Should().BeOfType(typeof(string));
```
Comment (aivascu) [inline NPlaylist/Tests/NPlaylist.Business.Tests/Providers/PathProviderTests.cs]: separeate the Act part from the Assert

File: NPlaylist/NPlaylist.Infrastructure/System/PathWrapper.cs [hunk 5]
```diff
@@ -13,5 +14,10 @@ public string GetExtension(string fileName)
         {
             return Path.GetExtension(fileName);
         }
+
+        public string Combine(string[] pathParts)
```
Comment (aivascu) [inline NPlaylist/NPlaylist.Infrastructure/System/PathWrapper.cs]: any reason you didn't provide the params option?

File: NPlaylist/Tests/NPlaylist.Business.Tests/Providers/PathProviderTests.cs (hunk 4 above)
Comment (Slickoo) [reply NPlaylist/Tests/NPlaylist.Business.Tests/Providers/PathProviderTests.cs]: smth like this?
~~~C#
var actualResult = sut.BuildPath("test.mp3");
actualResult.Should().BeOfType(typeof(string));
~~~

**Pull request merged**